extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/health/MongoHealthCheck.java

```diff
@@ -17,40 +16,78 @@
 import com.mongodb.client.MongoClient;
 
 import io.quarkus.arc.Arc;
+import io.quarkus.mongodb.MongoClientName;
+import io.quarkus.mongodb.reactive.ReactiveMongoClient;
 
 @Readiness
 @ApplicationScoped
 public class MongoHealthCheck implements HealthCheck {
-    private static final String DEFAULT_CLIENT = "__default__";
+
     private Map<String, MongoClient> clients = new HashMap<>();
+    private Map<String, ReactiveMongoClient> reactiveClients = new HashMap<>();
 
     @PostConstruct
     protected void init() {
-        Set<Bean<?>> beans = Arc.container().beanManager().getBeans(MongoClient.class);
-        for (Bean<?> bean : beans) {
+        for (Bean<?> bean : Arc.container().beanManager().getBeans(MongoClient.class)) {
```

Comment:
@mkouba wondering if there is an ArC API allowing to avoid the `BeanManager` call then the individual ArC calls below?

Looks like `select` is not "cross-qualifiers" so wouldn't get all the instances or am I wrong?

Reply:
I'm not quite sure what "cross-qualifiers" means but you can do `Arc.container().select(MongoClient.class, Any.Literal.INSTANCE)` to obtain all beans that satisfy the `MongoClient` required type no matter what other qualifiers are defined. But then in order to be able to access the qualifiers for each bean instance you would have to iterate over the instance handles, i.e. something like:
```java
for (InstanceHandle<MongoClient> handle : Arc.container().select(MongoClient.class, Any.Literal.INSTANCE).handles()) {
     String clientName = getMongoClientName(handle.getBean());
     .... // I'm not sure about the bean.getName() part?
     clients.put(clientName == null ? "default" : clientName, handle.get());
}
```
In any case, the `(MongoClient) Arc.container().instance(bean.getName()).get()` part does not seem to be very safe...

Reply:
The `(MongoClient) Arc.container().instance(bean.getName()).get()` came really from previous implementation.

Reply:
@mkouba If I remember correctly when there is multiple beans of the same type, only one can have no name (so `bean.getName() return null) and the other will have a name. At least for MongoDB this is how we create them (at least when we created the health check).

This is also how it is done on the Datasource extension (OK, both are my impl so it's normal): https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java#L28

If we change it here, we may change it in all extensions that have such health checks.

Reply:
> If I remember correctly when there is multiple beans of the same type, only one can have no name...

@loicmathieu Hm, only one can have the `@Default` qualifier (i.e. a bean does not explicitly declare a qualifier other than `@Named` or `@Any`). `bean.getName()` merely returns the name if it's annotated with `@Named`...

Reply:
So it only worked because we programmatically add the `@Named` annotation whene generating the named client inside the processor.
I checked on Agroal and this is also the case.

Reply:
I'll review the other health check impl to align with this one.

Reply:
Thanks @mkouba , that's exactly what I was asking. Good to know about `Any`.

@lkrzyzanek yeah, I know it was preexisting but it's a good opportunity to fix this! Thanks for making the changes.

Reply:
Thanks @mkouba @gsmet for raising this one. I think we'll need to do the same for the redis-client extension too. I'll do it in a separate PR sometime this week.